### PR TITLE
Bbcspa 1701 update modal ref

### PR DESCRIPTION
**Pull request by nickoola**

### CODE REVIEW

**Comment by nickoola** [comment]
Local tested and work on Kairos

File: src/modal/index.unit.tsx [hunk 1]
```diff
@@ -15,6 +15,12 @@ const defaultProps = {
 }
 
 describe('Modal', () => {
+  // TODO: mount methode will be fixed with forwardRef new feature in Enzyme v3.5
```
Comment (wakooka) [inline src/modal/index.unit.tsx]: nit: `method`

Comment (nickoola) [reply src/modal/index.unit.tsx]: Done!

File: src/modal/index.tsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ export interface ModalProps {
   readonly fullscreen?: boolean
   readonly displayDimmer?: boolean
   readonly closeButtonTitle?: string
+  readonly forwardedRef?: Ref<HTMLDivElement>
```
Comment (PerrineBoissieres) [inline src/modal/index.tsx]: Are you sure this is the right type? This seems pretty restrictive. Does `HTMLElement` work?

Comment (pith) [reply src/modal/index.tsx]: It's not restrictive, it the right type as the ref is applied on a div.

**Comment by pith** [comment]
You need to resolve conflicts before we can approve ;)

File: src/modal/index.unit.tsx [hunk 3]
```diff
@@ -15,6 +15,12 @@ const defaultProps = {
 }
 
 describe('Modal', () => {
+  // TODO: mount method will be fixed with forwardRef new feature in Enzyme v3.5
+  // As said in this post:
+  // https://www.bountysource.com/issues/
+  // 56657445-forwardref-enzyme-internal-error-unknown-node-with-tag-14
+
+/*
```
Comment (wakooka) [inline src/modal/index.unit.tsx]: Enzyme 3.5 is out, you can update package.json, and uncomment the test
https://github.com/airbnb/enzyme/blob/master/CHANGELOG.md

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,7 @@
 - **[UPDATE]** `IncreaseIcon` and `DecreaseIcon` are now `PlusIcon` and `MinusIcon`
 - **[UPDATE]** `TripCard` is now using `Text` component
 - **[FIX]** Margin between the elements on the bottom part of the `TripCard` when resizing
+- **[UPDATE]** Add ref attribute on Modal using forwardRef new API
```
Comment (Bidjii) [inline CHANGELOG.md]: Maybe in the `Unreleased` part? :)

Comment (nickoola) [reply CHANGELOG.md]: Right!

Comment (nickoola) [reply CHANGELOG.md]: Done!

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
 # Unreleased
-- [...]
+- **[UPDATE]** Modal use forwardRef new API
```
[the remark this answers is not in the document]
Comment (nickoola) [reply CHANGELOG.md]: Done!

**Pull request merged**